core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -142,10 +142,20 @@ public void available() {
                 final boolean isSystemDatabase = db.databaseName().equals(GraphDatabaseSettings.SYSTEM_DATABASE_NAME);
                 Configuration config = dependencyResolver.resolveDependency(ApocConfig.class).getConfig();
                 var initializers = collectInitializers(isSystemDatabase, config);
+
+                // Wait for database to become available, once it is available
+                // procedures are also available by necessity.
                 if (!isSystemDatabase) {
                     awaitApocProceduresRegistered();
-                } else if (!initializers.isEmpty()) {
-                    waitForSystemDbLifecycles();
+                } else {
+                    if (!initializers.isEmpty()) {
+                        waitForSystemDbLifecycles();
+                    }
+                }
+
+                // Register listener to drop apoc nodes from system databsae once a database is deleted
+                if (isSystemDatabase) {
+                    databaseEventListeners.registerDatabaseEventListener(new SystemFunctionalityListener());
                 }
```

Comment:
This can be moved into the else {...} above instead as that is following `if (!isSystemDatabase)`. That way we don't have to do the check again